<Type Name="AccessViolationException" FullName="System.AccessViolationException">
  <Metadata><Meta Name="ms.openlocfilehash" Value="af4cae868333721997233f2ac2270d3287455362" /><Meta Name="ms.sourcegitcommit" Value="2767b5903996fbd7e2b81eda1114dfc9d0d9db63" /><Meta Name="ms.translationtype" Value="MT" /><Meta Name="ms.contentlocale" Value="ru-RU" /><Meta Name="ms.lasthandoff" Value="05/02/2019" /><Meta Name="ms.locfileid" Value="65009441" /></Metadata><TypeSignature Language="C#" Value="public class AccessViolationException : SystemException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable beforefieldinit AccessViolationException extends System.SystemException" />
  <TypeSignature Language="DocId" Value="T:System.AccessViolationException" />
  <TypeSignature Language="VB.NET" Value="Public Class AccessViolationException&#xA;Inherits SystemException" />
  <TypeSignature Language="C++ CLI" Value="public ref class AccessViolationException : SystemException" />
  <TypeSignature Language="F#" Value="type AccessViolationException = class&#xA;    inherit SystemException" />
  <AssemblyInfo>
    <AssemblyName>mscorlib</AssemblyName>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
    <AssemblyVersion>2.0.5.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>netstandard</AssemblyName>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
    <AssemblyVersion>2.1.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>System.Runtime</AssemblyName>
    <AssemblyVersion>4.2.0.0</AssemblyVersion>
    <AssemblyVersion>4.2.1.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.SystemException</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute FrameworkAlternate="netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0;netframework-4.8">
      <AttributeName>System.Runtime.InteropServices.ComVisible(true)</AttributeName>
    </Attribute>
    <Attribute FrameworkAlternate="netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0">
      <AttributeName>System.Serializable</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary><span data-ttu-id="10b8a-101">Исключение, возникающее при попытке чтения или записи в защищенную область памяти.</span><span class="sxs-lookup"><span data-stu-id="10b8a-101">The exception that is thrown when there is an attempt to read or write protected memory.</span></span></summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="10b8a-102">Нарушение прав доступа в неуправляемом или небезопасном коде при попытке чтения или записи в память, которая не была выделена или к которой нет доступа.</span><span class="sxs-lookup"><span data-stu-id="10b8a-102">An access violation occurs in unmanaged or unsafe code when the code attempts to read or write to memory that has not been allocated, or to which it does not have access.</span></span> <span data-ttu-id="10b8a-103">Обычно это происходит, поскольку указатель имеет неверное значение.</span><span class="sxs-lookup"><span data-stu-id="10b8a-103">This usually occurs because a pointer has a bad value.</span></span> <span data-ttu-id="10b8a-104">Не все операции чтения или записи по неправильным указателям приводят к нарушениям прав доступа, поэтому нарушение прав доступа обычно указывает, что таких операций было несколько и что память может быть повреждена.</span><span class="sxs-lookup"><span data-stu-id="10b8a-104">Not all reads or writes through bad pointers lead to access violations, so an access violation usually indicates that several reads or writes have occurred through bad pointers, and that memory might be corrupted.</span></span> <span data-ttu-id="10b8a-105">Таким образом нарушение прав доступа почти всегда указывают на серьезные ошибки программирования.</span><span class="sxs-lookup"><span data-stu-id="10b8a-105">Thus, access violations almost always indicate serious programming errors.</span></span> <span data-ttu-id="10b8a-106"><xref:System.AccessViolationException> Четко определяет этих серьезных ошибок.</span><span class="sxs-lookup"><span data-stu-id="10b8a-106">An <xref:System.AccessViolationException> clearly identifies these serious errors.</span></span>  
  
 <span data-ttu-id="10b8a-107">В программах, состоящий только из проверяемых управляемого кода все ссылки являются либо допустимыми, либо значение null, и нарушения прав доступа невозможны.</span><span class="sxs-lookup"><span data-stu-id="10b8a-107">In programs consisting entirely of verifiable managed code, all references are either valid or null, and access violations are impossible.</span></span> <span data-ttu-id="10b8a-108">Любая операция, которая пытается сослаться на пустую ссылку в проверяемом коде вызывает исключение <xref:System.NullReferenceException> исключение.</span><span class="sxs-lookup"><span data-stu-id="10b8a-108">Any operation that attempts to reference a null reference in verifiable code throws a <xref:System.NullReferenceException> exception.</span></span> <span data-ttu-id="10b8a-109"><xref:System.AccessViolationException> Происходит, только когда проверяемый управляемый код взаимодействует с неуправляемым кодом или с небезопасным управляемым кодом.</span><span class="sxs-lookup"><span data-stu-id="10b8a-109">An <xref:System.AccessViolationException> occurs only when verifiable managed code interacts with unmanaged code or with unsafe managed code.</span></span>  
  
## <a name="troubleshooting-accessviolationexception-exceptions"></a><span data-ttu-id="10b8a-110">AccessViolationException исключениями</span><span class="sxs-lookup"><span data-stu-id="10b8a-110">Troubleshooting AccessViolationException exceptions</span></span>  
 <span data-ttu-id="10b8a-111"><xref:System.AccessViolationException> Исключение может возникнуть только в небезопасном управляемом коде или при управлении проверяемый код взаимодействует с неуправляемым кодом:</span><span class="sxs-lookup"><span data-stu-id="10b8a-111">An <xref:System.AccessViolationException> exception can occur only in unsafe managed code or when verifiable managed code interacts with unmanaged code:</span></span>  
  
-   <span data-ttu-id="10b8a-112">Нарушение прав доступа, возникающее в небезопасном управляемом коде может быть выражено как <xref:System.NullReferenceException> исключение или <xref:System.AccessViolationException> исключения, в зависимости от платформы.</span><span class="sxs-lookup"><span data-stu-id="10b8a-112">An access violation that occurs in unsafe managed code can be expressed as either a <xref:System.NullReferenceException> exception or an <xref:System.AccessViolationException> exception, depending on the platform.</span></span>  
  
-   <span data-ttu-id="10b8a-113">Нарушение прав доступа в неуправляемом коде, поднимается вверх к управляемому коду всегда упаковывается в <xref:System.AccessViolationException> исключение.</span><span class="sxs-lookup"><span data-stu-id="10b8a-113">An access violation in unmanaged code that bubbles up to managed code is always wrapped in an <xref:System.AccessViolationException> exception.</span></span>  
  
 <span data-ttu-id="10b8a-114">В любом случае можно определить и устранить причину <xref:System.AccessViolationException> исключение следующим образом:</span><span class="sxs-lookup"><span data-stu-id="10b8a-114">In either case, you can identify and correct the cause of the <xref:System.AccessViolationException> exception as follows:</span></span>  
  
 <span data-ttu-id="10b8a-115">Убедитесь, что память, в которую производится доступ, была выделена.</span><span class="sxs-lookup"><span data-stu-id="10b8a-115">Make sure that the memory that you are attempting to access has been allocated.</span></span>  
 <span data-ttu-id="10b8a-116"><xref:System.AccessViolationException> Всегда исключение при попытке получить доступ к защищенной памяти — то есть, для доступа к памяти, который не выделен, или, не принадлежащие процессу.</span><span class="sxs-lookup"><span data-stu-id="10b8a-116">An <xref:System.AccessViolationException> exception is always thrown by an attempt to access protected memory -- that is, to access memory that is not allocated or that is not owned by a process.</span></span>  
  
 <span data-ttu-id="10b8a-117">Автоматическое управление памятью — это одна из служб, которые предоставляет среда CLR.</span><span class="sxs-lookup"><span data-stu-id="10b8a-117">Automatic memory management is one of the services that the common language runtime provides.</span></span> <span data-ttu-id="10b8a-118">Если управляемый код обеспечивает ту же функциональность, что ваш неуправляемый код, можно переместить в управляемый код, чтобы воспользоваться преимуществами этой функции.</span><span class="sxs-lookup"><span data-stu-id="10b8a-118">If managed code provides the same functionality as your unmanaged code, you may wish to move to managed code to take advantage of this functionality.</span></span> <span data-ttu-id="10b8a-119">Для получения дополнительной информации см. [Automatic Memory Management](~/docs/standard/automatic-memory-management.md).</span><span class="sxs-lookup"><span data-stu-id="10b8a-119">For more information, see [Automatic Memory Management](~/docs/standard/automatic-memory-management.md).</span></span>  
  
 <span data-ttu-id="10b8a-120">Убедитесь, что память, в которую производится доступ, не была повреждена.</span><span class="sxs-lookup"><span data-stu-id="10b8a-120">Make sure that the memory that you are attempting to access has not been corrupted.</span></span>  
 <span data-ttu-id="10b8a-121">Если несколько операций чтения или записи произошли по неправильным указателям, память может быть повреждена.</span><span class="sxs-lookup"><span data-stu-id="10b8a-121">If several read or write operations have occurred through bad pointers, memory may be corrupted.</span></span> <span data-ttu-id="10b8a-122">Обычно это происходит при чтении или записи для адресов за пределами предопределенные буфера.</span><span class="sxs-lookup"><span data-stu-id="10b8a-122">This typically occurs when reading or writing to addresses outside of a predefined buffer.</span></span>  
  
## <a name="accessviolationexception-and-trycatch-blocks"></a><span data-ttu-id="10b8a-123">Блоки AccessViolationException и try/catch</span><span class="sxs-lookup"><span data-stu-id="10b8a-123">AccessViolationException and try/catch blocks</span></span>  
 <span data-ttu-id="10b8a-124">Начиная с [!INCLUDE[net_v40_long](~/includes/net-v40-long-md.md)], <xref:System.AccessViolationException> исключения, создаваемые среда CLR не обрабатываются `catch` инструкции в структурированный обработчик исключений при возникновении исключения вне памяти, зарезервированной с помощью общего языка Среда выполнения.</span><span class="sxs-lookup"><span data-stu-id="10b8a-124">Starting with the [!INCLUDE[net_v40_long](~/includes/net-v40-long-md.md)], <xref:System.AccessViolationException> exceptions thrown by the common language runtime are not handled by the `catch` statement in a structured exception handler if the exception occurs outside of the memory reserved by the common language runtime.</span></span> <span data-ttu-id="10b8a-125">Для обработки таких <xref:System.AccessViolationException> исключение, необходимо применить <xref:System.Runtime.ExceptionServices.HandleProcessCorruptedStateExceptionsAttribute> атрибут к методу, в котором возникает исключение.</span><span class="sxs-lookup"><span data-stu-id="10b8a-125">To handle such an <xref:System.AccessViolationException> exception, you should apply the  <xref:System.Runtime.ExceptionServices.HandleProcessCorruptedStateExceptionsAttribute> attribute to the method in which the exception is thrown.</span></span> <span data-ttu-id="10b8a-126">Это изменение не влияет на <xref:System.AccessViolationException> исключения, создаваемые пользователем код, который можно продолжать перехватываться `catch` инструкции.</span><span class="sxs-lookup"><span data-stu-id="10b8a-126">This change does not affect <xref:System.AccessViolationException> exceptions thrown by user code, which can continue to be caught by a `catch` statement.</span></span> <span data-ttu-id="10b8a-127">Для кода, написанные для предыдущих версий платформы .NET Framework, необходимо перекомпилировать и выполняться без доработок и [!INCLUDE[net_v40_long](~/includes/net-v40-long-md.md)], вы можете добавить [ \<legacyCorruptedStateExceptionsPolicy >](~/docs/framework/configure-apps/file-schema/runtime/legacycorruptedstateexceptionspolicy-element.md) элемент к файл конфигурации этого приложения.</span><span class="sxs-lookup"><span data-stu-id="10b8a-127">For code written for previous versions of the .NET Framework that you want to recompile and run without modification on the [!INCLUDE[net_v40_long](~/includes/net-v40-long-md.md)], you can add the [\<legacyCorruptedStateExceptionsPolicy>](~/docs/framework/configure-apps/file-schema/runtime/legacycorruptedstateexceptionspolicy-element.md) element to your app's configuration file.</span></span> <span data-ttu-id="10b8a-128">Обратите внимание на то, что могут также получать уведомления о исключения, если вы определили обработчик для <xref:System.AppDomain.FirstChanceException?displayProperty=nameWithType> или <xref:System.AppDomain.UnhandledException?displayProperty=nameWithType> событий.</span><span class="sxs-lookup"><span data-stu-id="10b8a-128">Note that you can also receive notification of the exceptions if you have defined a handler for the <xref:System.AppDomain.FirstChanceException?displayProperty=nameWithType> or <xref:System.AppDomain.UnhandledException?displayProperty=nameWithType> event.</span></span>  
  
## <a name="version-information"></a><span data-ttu-id="10b8a-129">Сведения о версии</span><span class="sxs-lookup"><span data-stu-id="10b8a-129">Version Information</span></span>  
 <span data-ttu-id="10b8a-130">Это исключение возникает в .NET Framework 2.0 и более поздних версий.</span><span class="sxs-lookup"><span data-stu-id="10b8a-130">This exception is thrown in the .NET Framework 2.0 and later versions.</span></span> <span data-ttu-id="10b8a-131">В более ранних версиях платформы .NET Framework, соответствует нарушение прав доступа в неуправляемый код или небезопасный управляемый код <xref:System.NullReferenceException> в управляемом коде.</span><span class="sxs-lookup"><span data-stu-id="10b8a-131">In earlier versions of the .NET Framework, an access violation in unmanaged code or unsafe managed code is represented by a <xref:System.NullReferenceException> in managed code.</span></span> <span data-ttu-id="10b8a-132">Объект <xref:System.NullReferenceException> также возникает, когда пустая ссылка является разыменован в проверяемый управляемый код, вхождение, не связанных с повреждением данных, и нет способа для различения двух ситуаций в версии 1.0 или 1.1.</span><span class="sxs-lookup"><span data-stu-id="10b8a-132">A <xref:System.NullReferenceException> is also thrown when a null reference is dereferenced in verifiable managed code, an occurrence that does not involve data corruption, and there is no way to distinguish between the two situations in versions 1.0 or 1.1.</span></span>  
  
 <span data-ttu-id="10b8a-133">Администраторы могут позволить отдельным приложениям, чтобы вернуться к поведению в .NET Framework версии 1.1.</span><span class="sxs-lookup"><span data-stu-id="10b8a-133">Administrators can allow selected applications to revert to the behavior of the .NET Framework version 1.1.</span></span> <span data-ttu-id="10b8a-134">Добавьте следующую строку в [ &lt;среды выполнения&gt; элемент](~/docs/framework/configure-apps/file-schema/runtime/runtime-element.md) раздел файла конфигурации приложения:</span><span class="sxs-lookup"><span data-stu-id="10b8a-134">Place the following line in the [&lt;runtime&gt; Element](~/docs/framework/configure-apps/file-schema/runtime/runtime-element.md) section of the configuration file for the application:</span></span>  
  
```  
<legacyNullReferenceExceptionPolicy enabled = "1"/>  
```  
  
 ]]></format>
    </remarks>
    <related type="Article" href="~/docs/standard/exceptions/index.md"><span data-ttu-id="10b8a-135">Обработка и создание исключений</span><span class="sxs-lookup"><span data-stu-id="10b8a-135">Handling and Throwing Exceptions</span></span></related>
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary><span data-ttu-id="10b8a-136">Инициализирует новый экземпляр класса <see cref="T:System.AccessViolationException" />.</span><span class="sxs-lookup"><span data-stu-id="10b8a-136">Initializes a new instance of the <see cref="T:System.AccessViolationException" /> class.</span></span></summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public AccessViolationException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AccessViolationException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; AccessViolationException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary><span data-ttu-id="10b8a-137">Инициализирует новый экземпляр класса <see cref="T:System.AccessViolationException" /> с системным сообщением, содержащим описание ошибки.</span><span class="sxs-lookup"><span data-stu-id="10b8a-137">Initializes a new instance of the <see cref="T:System.AccessViolationException" /> class with a system-supplied message that describes the error.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="10b8a-138">Этот конструктор инициализирует <xref:System.Exception.Message%2A> свойства нового экземпляра системное сообщение с описанием ошибки, такие как «попытка чтения или записи в защищенную память.</span><span class="sxs-lookup"><span data-stu-id="10b8a-138">This constructor initializes the <xref:System.Exception.Message%2A> property of the new instance to a system-supplied message that describes the error, such as "Attempted to read or write protected memory.</span></span> <span data-ttu-id="10b8a-139">Это часто указывает на то что другие память повреждена.»</span><span class="sxs-lookup"><span data-stu-id="10b8a-139">This is often an indication that other memory has been corrupted."</span></span> <span data-ttu-id="10b8a-140">Это сообщение учитывает текущую культуру системы.</span><span class="sxs-lookup"><span data-stu-id="10b8a-140">This message takes into account the current system culture.</span></span>  
  
 <span data-ttu-id="10b8a-141">В следующей таблице представлены исходные значения свойств экземпляра класса <xref:System.AccessViolationException>.</span><span class="sxs-lookup"><span data-stu-id="10b8a-141">The following table shows the initial property values for an instance of <xref:System.AccessViolationException>.</span></span>  
  
|<span data-ttu-id="10b8a-142">Свойство.</span><span class="sxs-lookup"><span data-stu-id="10b8a-142">Property</span></span>|<span data-ttu-id="10b8a-143">Значение</span><span class="sxs-lookup"><span data-stu-id="10b8a-143">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="10b8a-144">`null`.</span><span class="sxs-lookup"><span data-stu-id="10b8a-144">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="10b8a-145">Строка локализованного сообщения об ошибке.</span><span class="sxs-lookup"><span data-stu-id="10b8a-145">The localized error message string.</span></span>|  
  
 ]]></format>
        </remarks>
        <related type="Article" href="~/docs/standard/exceptions/index.md"><span data-ttu-id="10b8a-146">Обработка и создание исключений</span><span class="sxs-lookup"><span data-stu-id="10b8a-146">Handling and Throwing Exceptions</span></span></related>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public AccessViolationException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AccessViolationException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; AccessViolationException(System::String ^ message);" />
      <MemberSignature Language="F#" Value="new AccessViolationException : string -&gt; AccessViolationException" Usage="new System.AccessViolationException message" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="10b8a-147">Сообщение с описанием исключения.</span><span class="sxs-lookup"><span data-stu-id="10b8a-147">The message that describes the exception.</span></span> <span data-ttu-id="10b8a-148">Код, вызывающий этот конструктор, должен обеспечить локализацию данной строки в соответствии с текущим языком и региональными параметрами системы.</span><span class="sxs-lookup"><span data-stu-id="10b8a-148">The caller of this constructor is required to ensure that this string has been localized for the current system culture.</span></span></param>
        <summary><span data-ttu-id="10b8a-149">Инициализирует новый экземпляр класса <see cref="T:System.AccessViolationException" /> с использованием заданного сообщения, содержащего описание ошибки.</span><span class="sxs-lookup"><span data-stu-id="10b8a-149">Initializes a new instance of the <see cref="T:System.AccessViolationException" /> class with a specified message that describes the error.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="10b8a-150">В следующей таблице представлены исходные значения свойств экземпляра класса <xref:System.AccessViolationException>.</span><span class="sxs-lookup"><span data-stu-id="10b8a-150">The following table shows the initial property values for an instance of <xref:System.AccessViolationException>.</span></span>  
  
|<span data-ttu-id="10b8a-151">Свойство.</span><span class="sxs-lookup"><span data-stu-id="10b8a-151">Property</span></span>|<span data-ttu-id="10b8a-152">Значение</span><span class="sxs-lookup"><span data-stu-id="10b8a-152">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="10b8a-153">`null`.</span><span class="sxs-lookup"><span data-stu-id="10b8a-153">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="10b8a-154">Строка сообщения об ошибке, указанная в `message`.</span><span class="sxs-lookup"><span data-stu-id="10b8a-154">The error message string specified in `message`.</span></span>|  
  
 ]]></format>
        </remarks>
        <related type="Article" href="~/docs/standard/exceptions/index.md"><span data-ttu-id="10b8a-155">Обработка и создание исключений</span><span class="sxs-lookup"><span data-stu-id="10b8a-155">Handling and Throwing Exceptions</span></span></related>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="protected AccessViolationException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method familyhidebysig specialname rtspecialname instance void .ctor(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AccessViolationException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Protected Sub New (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="protected:&#xA; AccessViolationException(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberSignature Language="F#" Value="new AccessViolationException : System.Runtime.Serialization.SerializationInfo * System.Runtime.Serialization.StreamingContext -&gt; AccessViolationException" Usage="new System.AccessViolationException (info, context)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Security.SecuritySafeCritical</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info"><span data-ttu-id="10b8a-156">Класс <see cref="T:System.Runtime.Serialization.SerializationInfo" />, содержащий сериализованные данные объекта.</span><span class="sxs-lookup"><span data-stu-id="10b8a-156">The <see cref="T:System.Runtime.Serialization.SerializationInfo" /> that holds the serialized object data.</span></span></param>
        <param name="context"><span data-ttu-id="10b8a-157">Объект <see cref="T:System.Runtime.Serialization.StreamingContext" />, содержащий контекстные сведения об источнике или назначении.</span><span class="sxs-lookup"><span data-stu-id="10b8a-157">The <see cref="T:System.Runtime.Serialization.StreamingContext" /> that contains contextual information about the source or destination.</span></span></param>
        <summary><span data-ttu-id="10b8a-158">Инициализирует новый экземпляр класса <see cref="T:System.AccessViolationException" /> с сериализованными данными.</span><span class="sxs-lookup"><span data-stu-id="10b8a-158">Initializes a new instance of the <see cref="T:System.AccessViolationException" /> class with serialized data.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="10b8a-159">Данный конструктор вызывается в процессе десериализации для воссоздания объекта исключения, переданного в потоке.</span><span class="sxs-lookup"><span data-stu-id="10b8a-159">This constructor is called during deserialization to reconstitute the exception object transmitted over a stream.</span></span> <span data-ttu-id="10b8a-160">Дополнительные сведения см. в разделе [сериализация XML и SOAP](~/docs/standard/serialization/xml-and-soap-serialization.md).</span><span class="sxs-lookup"><span data-stu-id="10b8a-160">For more information, see [XML and SOAP Serialization](~/docs/standard/serialization/xml-and-soap-serialization.md).</span></span>  
  
 ]]></format>
        </remarks>
        <related type="Article" href="~/docs/standard/serialization/xml-and-soap-serialization.md"><span data-ttu-id="10b8a-161">Сериализация XML и SOAP</span><span class="sxs-lookup"><span data-stu-id="10b8a-161">XML and SOAP Serialization</span></span></related>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public AccessViolationException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AccessViolationException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; AccessViolationException(System::String ^ message, Exception ^ innerException);" />
      <MemberSignature Language="F#" Value="new AccessViolationException : string * Exception -&gt; AccessViolationException" Usage="new System.AccessViolationException (message, innerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="10b8a-162">Сообщение с описанием исключения.</span><span class="sxs-lookup"><span data-stu-id="10b8a-162">The message that describes the exception.</span></span> <span data-ttu-id="10b8a-163">Код, вызывающий этот конструктор, должен обеспечить локализацию данной строки в соответствии с текущим языком и региональными параметрами системы.</span><span class="sxs-lookup"><span data-stu-id="10b8a-163">The caller of this constructor is required to ensure that this string has been localized for the current system culture.</span></span></param>
        <param name="innerException"><span data-ttu-id="10b8a-164">Исключение, которое является причиной текущего исключения.</span><span class="sxs-lookup"><span data-stu-id="10b8a-164">The exception that is the cause of the current exception.</span></span> <span data-ttu-id="10b8a-165">Если значение параметра <paramref name="innerException" /> не равно <see langword="null" />, текущее исключение сгенерировано в блоке <see langword="catch" />, обрабатывающем внутреннее исключение.</span><span class="sxs-lookup"><span data-stu-id="10b8a-165">If the <paramref name="innerException" /> parameter is not <see langword="null" />, the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span></span></param>
        <summary><span data-ttu-id="10b8a-166">Инициализирует новый экземпляр класса <see cref="T:System.AccessViolationException" /> указанным сообщением об ошибке и ссылкой на внутреннее исключение, вызвавшее данное исключение.</span><span class="sxs-lookup"><span data-stu-id="10b8a-166">Initializes a new instance of the <see cref="T:System.AccessViolationException" /> class with a specified error message and a reference to the inner exception that is the cause of this exception.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="10b8a-167">Исключение, созданное как прямой результат предыдущего исключения, должно содержать в свойстве <xref:System.Exception.InnerException%2A> ссылку на предыдущее исключение.</span><span class="sxs-lookup"><span data-stu-id="10b8a-167">An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="10b8a-168">Свойство <xref:System.Exception.InnerException%2A> возвращает значение, переданное конструктору, или значение `null`, если свойство <xref:System.Exception.InnerException%2A> не передает конструктору значение внутреннего исключения.</span><span class="sxs-lookup"><span data-stu-id="10b8a-168">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or `null` if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 <span data-ttu-id="10b8a-169">В следующей таблице представлены исходные значения свойств экземпляра класса <xref:System.AccessViolationException>.</span><span class="sxs-lookup"><span data-stu-id="10b8a-169">The following table shows the initial property values for an instance of <xref:System.AccessViolationException>.</span></span>  
  
|<span data-ttu-id="10b8a-170">Свойство.</span><span class="sxs-lookup"><span data-stu-id="10b8a-170">Property</span></span>|<span data-ttu-id="10b8a-171">Значение</span><span class="sxs-lookup"><span data-stu-id="10b8a-171">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="10b8a-172">`null`.</span><span class="sxs-lookup"><span data-stu-id="10b8a-172">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="10b8a-173">Строка сообщения об ошибке, указанная в `message`.</span><span class="sxs-lookup"><span data-stu-id="10b8a-173">The error message string specified in `message`.</span></span>|  
  
 ]]></format>
        </remarks>
        <related type="Article" href="~/docs/standard/exceptions/index.md"><span data-ttu-id="10b8a-174">Обработка и создание исключений</span><span class="sxs-lookup"><span data-stu-id="10b8a-174">Handling and Throwing Exceptions</span></span></related>
      </Docs>
    </Member>
  </Members>
</Type>